Add issue sensors to Overseerr integration

AmGarera opened the pull request:
## Description

This PR adds 6 new sensors to track issue statistics in Overseerr. This is the first part of comprehensive issue management functionality - service actions will be added in a separate PR per review guidelines.

**Note:** This PR is now based on #158924 and depends on it being merged first.

## Changes

### New Sensors
- Total issues
- Open issues  
- Closed issues
- Video issues
- Audio issues
- Subtitle issues

### Technical Changes
- Created models.py with OverseerrData composite model containing both RequestCount and IssueCount
- Updated coordinator.py to fetch both request and issue counts in single update cycle
- Modified sensor.py to use composite data model (data.requests.* and data.issues.*)
- Added ISSUES constant to const.py
- Added webhook support for issue notification types (ISSUE_REPORTED, ISSUE_COMMENTED, ISSUE_RESOLVED, ISSUE_REOPENED)
- Added sensor translations to strings.json

## Dependencies

- Depends on #158924 (python-overseerr 0.8.0 bump)
- Requires python-overseerr 0.8.0 (PR: https://github.com/joostlek/python-overseerr/pull/291) ✅ Merged

## Type of change

- [ ] New integration
- [x] New feature for existing integration
- [ ] Bugfix for existing integration  
- [ ] Breaking change
- [ ] Code quality improvements

## Additional information

- Services for issue management will be added in a follow-up PR
- All new code follows Platinum quality scale requirements
- Test coverage maintained above 95%

## Checklist

- [x] The code change is tested and works locally
- [x] Tests have been added to verify the new code works
- [x] Documentation has been updated (separate PR: https://github.com/home-assistant/home-assistant.io/pull/42494)

## Related Issues/PRs

- python-overseerr enhancement: https://github.com/joostlek/python-overseerr/pull/291 ✅ Merged
- Dependency bump: #158924 (merge this first)
- Documentation: https://github.com/home-assistant/home-assistant.io/pull/42494
- Replaces combined PR #158512 (split per review feedback)

AmGarera committed:
Add issue sensors to Overseerr integration

Add 6 new sensors to track issues in Overseerr:
- Total issues
- Open issues
- Closed issues
- Video issues
- Audio issues
- Subtitle issues

Update coordinator to fetch both request and issue counts.
Add webhook support for issue notifications.

joostlek reviewed (changes requested):
Can we add tests for this?

AmGarera committed:
Add comprehensive coordinator tests for Overseerr

Add test coverage for the OverseerrCoordinator including:
- Successful data fetch for both requests and issues
- Authentication error handling
- Connection error handling
- Data refresh functionality
- Proper exception raising (ConfigEntryAuthFailed, UpdateFailed)

Also add issue_count.json test fixture and update conftest to mock
get_issue_count API method.

AmGarera committed:
Fix coordinator test assertions to match fixture data

Update test assertions to use actual values from request_count.json
fixture instead of hardcoded incorrect values.

AmGarera committed:
Update tests for issue notification support

- Update REGISTERED_NOTIFICATIONS value from 222 to 4062 (now includes issue notifications)
- Remove 'Bigger scope' test case as we now use all notification types
- Update webhook_config.json fixture with new notification types value
- Update snapshots for new issue sensors
- All 52 tests passing with 100% coverage

AmGarera committed:
Fix sensor snapshot names for issue sensors

Replace placeholder 'none' entity IDs with proper translation-key-based IDs:
- sensor.overseerr_none → sensor.overseerr_total_issues
- sensor.overseerr_none_2 → sensor.overseerr_open_issues
- sensor.overseerr_none_3 → sensor.overseerr_closed_issues
- sensor.overseerr_none_4 → sensor.overseerr_video_issues
- sensor.overseerr_none_5 → sensor.overseerr_audio_issues
- sensor.overseerr_none_6 → sensor.overseerr_subtitle_issues

This fixes CI test failures on Python 3.14.2 where snapshots were not found.

joostlek commented on homeassistant/components/overseerr/sensor.py:
```diff
@@ -24,51 +23,87 @@
+    ),
+    OverseerrSensorEntityDescription(
+        key="total_issues",
+        native_unit_of_measurement=ISSUES,
```
Oh we can move the unit of measurement to the translations and then we can translate it :)

joostlek reviewed (changes requested):
Feel free to message me on Discord btw

AmGarera committed:
Use translatable units for issue sensors and add webhook test

- Move unit_of_measurement from hardcoded ISSUES constant to translation strings
- Remove ISSUES constant from const.py (no longer needed)
- Add 'state.measurement' translations for all 6 issue sensors
- Remove direct coordinator tests (test via state machine instead)
- Add webhook test for issue events (test_webhook_issue_trigger_update)
- Fix webhook handler to refresh coordinator for REQUEST_ and ISSUE_ events
- Add webhook_issue_reported.json test fixture
- Update snapshots for new unit_of_measurement approach

This allows the unit 'issues' to be translated to different languages.
